### DISCUSSION TITLE
Introducing Vector Protocol / Grant Proposal for a Vector Database

### DISCUSSION THREAD

**Discussion by JasonMarkWomack** [akash-network/community#430] [Governance Proposals]
**Overview**
Low Code Vector Database with GUI on Akash Network with CRUD like features

We propose to build Vector Protocol, Low Code Vector Database with GUI on Akash Network with CRUD like features, for Akash that will allow for vector embedding storage in sized to scale vector databases. Also, Vector Protocol enables multi-modal search operations. 
semantic search vs classification vs clustering vs association analysis

**Background**
The core of technical team has experience building web2 and web3 applications. 
CEO / Founder Jason Womack, A former Lead Software Engineer (AI/ML) Consultant with extensive experience at Raytheon, Asurion, P&G, and various venture-backed startups.

**Tech Info**
We plan to use Langchain, https://ollama.ai/, Golang, D3.js and Python. We plan to build our own vector store from scratch. 

I keep running into errors when using Pinecone. They have troublesome Upsert tooling/ features, among many other issues.  I want to build a better development tool that works with Open AI and connects to various LLMs.

To be more specific, our interface logic with LLMs allows for scalable DB storage and persistence across AI native software apps.  We have an in-house integrations framework extending capabilities up the stack to ensure devs can do CRUD-like database operations in a browser GUI. This means, they can use our low code GUI in a “Canva-like” way in order to build out their schema, define API endpoints, visually debug indices, configure long term memory and much more. .

Funding/Budget/Milestones 
Per the grant, we would like to apply for a $50K USD grant to build and maintain the the vector database

estimated hours to get to MVP  where it can hold multiple devs. 

70 USD  X 500 hours = $50K 

Contact
Email: [womackjason99@gmail.com]

**Comment by anilmurty** [comment]
Thanks for submitting this proposal Jason. Some things that would be helpful (that past successful proposal have had):
1. Some more background on the solution you are building (like "why does a vector DB matter and why does it need to be low-code?" specifically for people in the community who may not be as well versed with the topic as you are)
2. A basic POC if you have one already or at least some wireframes of the UX
3. A more detailed breakdown of the milestones leading up to (in this case) the MVP
4. Breakdown of funds needed for engneering/development vs compute resources
5. Plan for how you intend to attract users.
6. Ideally building your UX on top of cloudmos similar to what https://github.com/orgs/akash-network/discussions/406 is doing.

Just my initial feedback - would wait for others to also comment. 

Lastly, in terms of process, we typically wait for a few weeks of discussions and if there is enough interest, have a live discussion at the next steering committee where you'll have the chance to present this yourself (including any early ideation/ demos if you have them)

**Comment by piber-dev** [comment]
hello @JasonMarkWomack, 

do you have any information on how this will scale in terms of cost to the user and redundancy / reliability? akash is not primarily a storage chain and it might make more sense to e.g., simply port something that already works such as: a scalable [renterd](https://sia.tech/software/renterd) (PS: they have an active hackathon if you want to get involved and possibly earn prize money), or a [database using IPFS](https://github.com/orbitdb/orbitdb).

i like the fact that you want to build a scalable database on akash however we need more information (and possibly discussion) before this would be ready for akash. there are a lot of technical and practical details that you may not be aware of if you haven't used akash before regarding e.g., reliability and scaling. it is also important to know how much a typical database of X resources/configurations would cost to operate as a tenant.
